### DISCUSSION TITLE
tests in same file in V

### DISCUSSION THREAD

**Discussion by prashanth-hegde** [vlang/v#18746] [Ideas]
Wouldn't it be nice to write tests in the same file as code in V also.
Reference: [Zig](https://ziglang.org/documentation/master/#Zig-Test), 

Definitely not proposing to change the current way of test suites as shown here.
But proposing supporting another way of testing with test cases defined in the same file. We can have the same requirements as current, viz: 

* test functions need to start with `test_` name
* test cases are only executed when run in `v test .` command
* During build times, these tests are no-op

Such type of testing may be very beneficial for code that has a single file, or for shell scripts

**Comment by JalonSolov** [comment]
It has been suggested before, but turned down.

V looks for files with `_test.v` as the last part of the name.  This is how it keeps tests separate from your regular code.

**Comment by Wajinn** [comment]
We have to be careful about bringing in practices from different languages.  What some other language does, may not fit in another language for both technical and cultural reasons.

**Comment by prashanth-hegde** [comment]
Closing because it's been discussed before
